### PR TITLE
objfw: move passthru test to its own file 

### PR DESCRIPTION
**Pull request by steeleduncan**
Following up on a suggestion by @khaneliman in https://github.com/NixOS/nixpkgs/pull/366484#discussion_r1895244048, this splits the test into its own file

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ob/objfw/test-build-and-run.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+clangStdenv: objfw: writeTextDir:
```
Comment (khaneliman) [inline pkgs/by-name/ob/objfw/test-build-and-run.nix:1]:
```suggestion
{
clangStdenv,
objfw,
writeTextDir
}:
```

File: pkgs/by-name/ob/objfw/package.nix [hunk 2]
```diff
@@ -33,44 +33,7 @@ clangStdenv.mkDerivation (finalAttrs: {
...
-      '';
-      doCheck = true;
-
-      installPhase = ''
-        runHook preInstall
-        touch $out
-        runHook postInstall
-      '';
-    };
+    build-hello-world = (import ./test-build-and-run.nix) clangStdenv objfw writeTextDir;
```
Comment (khaneliman) [inline pkgs/by-name/ob/objfw/package.nix:36]:
```suggestion
    build-hello-world = import ./test-build-and-run.nix) { inherit clangStdenv objfw writeTextDir; };
```

**Review by khaneliman — COMMENTED** [review]
This would probably be cleaner and make it easier to adjust arguments like a normal derivation.

File: pkgs/by-name/ob/objfw/package.nix (hunk 2 above)
Comment (steeleduncan) [reply pkgs/by-name/ob/objfw/package.nix]: Thanks, I had a feeling there must be syntax along those lines, but couldn't find an example!

**Comment by khaneliman** [comment]
Any idea what is being suggested here https://github.com/NixOS/nixpkgs/pull/366484#discussion_r1895850944 ?

**Review by khaneliman — APPROVED** [review]

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367625`

**Comment by steeleduncan** [comment]
> Any idea what is being suggested here [#366484 (comment)](https://github.com/NixOS/nixpkgs/pull/366484#discussion_r1895850944) ?

Yes, there is an `objfw-config` tool that can generate compile options (similar to `pkgconfig`). I don't personally use it, so I didn't think of it, but there is a FHS assumption in there I'll need to patch to get it working, then I can use it for the test

**Comment by steeleduncan** [comment]
Thanjks for merging!

**Pull request merged**